packages/drivers/odsp-driver/src/fetchWithEpochValidation.ts

```diff
@@ -0,0 +1,125 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { ITelemetryLogger } from "@fluidframework/common-definitions";
+import { createOdspNetworkError, fluidEpochMismatchError, OdspErrorType } from "@fluidframework/odsp-doclib-utils";
+import { fetchAndParseAsJSONHelper, fetchHelper, IOdspResponse } from "./odspUtils";
+import { ICacheEntry, IPersistedCache, IPersistedCacheValue } from "./odspCache";
+
+/**
+ * This class is a wrapper around fetch calls. It adds epoch to the request made so that the
+ * server can match it with its epoch value in order to match the version.
+ * It also validates the epoch value received in response of fetch calls. If the epoch does not match,
+ * then it also clears all the cached entries for the given container.
+ */
+export class FetchWithEpochValidation {
+    private _fluidEpoch: string | undefined;
+    private _hashedDocumentId: string | undefined;
+    constructor(
+        private readonly persistedCache: IPersistedCache,
+        private readonly logger?: ITelemetryLogger) {
+    }
+
+    public set hashedDocumentId(docId: string | undefined) {
+        this._hashedDocumentId = docId;
```

Comment:
Ideally, we pass docId in constructor. Is this possible?
If not, I think we should not allow undefined here, and we should allow only calling this API once, i.e. only if we do not have doc ID set yet.

Reply:
It will not be passed in constructor because it is created before docid is assigned. I have added assert so as to make sure it is called only once and that passed docid is not undefined. However type would be string and undefined because getter and setter should be of same type.